gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py

```diff
@@ -0,0 +1,159 @@
+        return agent, sc
+
+    def _test_event_creation(self, method):
+        """ Tests event creation and enqueueing for create/delete
+        operation of generic config agent's RPC manager.
+
+        :param method: CREATE_NFP_SERVICE
+
+        Returns: none
+
+        """
+
+        agent, sc = self._get_NfpServiceRpcManager_object()
+        arg_dict = {'context': self.fo.context,
+                    'resource_data': self.fo.kwargs}
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event:
+            actual_call = agent.run_nfp_service(self.fo.context,
+                                                self.fo.kwargs)
+
+            expected_cal = mock_sc_event.assert_called_with(
+                                id=method, data=arg_dict, key=None)
+            self.assertEqual(actual_call, expected_cal)
+            mock_sc_rpc_event.assert_called_with('foo')
+
+    def test_nfp_service_rpcmanager(self):
+        """ Implements test case for run_nfp_service method
+        of nfp service RPC manager.
+
+        Returns: none
+
+        """
+
+        self._test_event_creation(const.CREATE_NFP_SERVICE_EVENT)
+
+
+class NfpServiceEventHandlerTestCase(base.BaseTestCase):
+    """ Implements test cases for event handler methods
+    of nfp service agent.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(NfpServiceEventHandlerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = fo.FakeObjects()
+        self.context = {'notification_data': {},
+                        'resource': 'interfaces'}
+
+    @mock.patch(__name__ + '.fo.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.fo.FakeObjects.drivers')
+    @mock.patch(__name__ + '.fo.FakeObjects.sc')
+    def _get_NfpServiceEventHandler_object(self, sc, drivers, rpcmgr):
+        """ Retrieves event handler object of nfp service.
+
+        :param sc: mocked service controller object of process model framework
+        :param rpcmgr: object of configurator's RPC manager
+        :param drivers: list of driver objects for nfp service agent
+
+        Returns: object of nfp service's event handler
+
+        """
+
+        agent = ns.ConfigScriptEventHandler(sc, drivers, rpcmgr)
+        return agent, sc
+
+    def _test_handle_event(self, ev, result=const.UNHANDLED_RESULT):
+        """ Test handle event method of nfp service agent.
+
+        :param ev: event data which has to be actually sent by
+        process framework.
+
+        Returns: None
+
+        """
+
+        agent, sc = self._get_NfpServiceEventHandler_object()
+        driver = mock.Mock()
+
+        with mock.patch.object(
+                driver, 'run_heat', return_value=result) as mock_config_inte, (
+             mock.patch.object(
+                agent, '_get_driver', return_value=driver)):
+
```

Comment:
Avoid too many spaces like this in the code. Not sure if PEP will complain, but keep one line gap which should be enough.